src/external_task_api_service.ts

```diff
@@ -129,6 +128,43 @@ export class ExternalTaskApiService implements IExternalTaskApi {
     this._publishExternalTaskFinishedMessage(externalTask, successNotificationPayload);
   }
 
+  private async _fetchOrWaitForExternalTasks<TPayloadType>(topicName: string, maxTasks: number, longPollingTimeout: number)
+    : Promise<Array<ExternalTask<TPayloadType>>> {
+
+    return new Promise<Array<ExternalTask<TPayloadType>>>(async (resolve): Promise<void> => {
+
+      const tasks: Array<ExternalTask<TPayloadType>> =
+        await this._externalTaskRepository.fetchAvailableForProcessing<TPayloadType>(topicName, maxTasks);
+
+      if (tasks.length > 0) {
+        resolve(tasks);
+      }
+      let subscription: ISubscription;
```

Comment:
I would make this `const` and directly assign it like so: 
```suggestion
      const externalTaskCreatedEventName = `/externaltask/topic/${topicName}/created`;
      const subscription: ISubscription = this._eventAggregator.subscribeOnce(externalTaskCreatedEventName, async (): Promise<void> => {
         clearTimeout(timeout);
         const tasks: Array<ExternalTask<TPayloadType>> =
          await this._externalTaskRepository.fetchAvailableForProcessing<TPayloadType>(topicName, maxTasks);
         if (tasks.length > 0) {
          resolve(tasks);
        } else {
          resolve([]);
        }
      })
    });
```

Reply:
This is not posssible. I need to define the subscription variable to use it in the setTimeout callback.

Reply:
I missed that we actually need to access `timeout` in the defined subscription.